src/api/browse.ts

```diff
@@ -10,43 +33,32 @@ type getDatasetsSearchRespType = (
   documentType: string
 ) => void;
 
-const postHeaders = {
-  Accept: "application/json",
-  "Content-Type": "application/json",
-  Origin: "http://localhost:8000/",
-};
-
-export const querySearchService: getDatasetsSearchRespType = (
+export const querySearchService: getDatasetsSearchRespType = async (
   callbackFunc: (hits: searchResponseModel) => void,
   filterQuery: facetFilterModel[],
   searchKeyword = "*",
   skip: number,
   limit: number,
   documentType = "Dataset"
 ) => {
-  fetch(
-    `${process.env.REACT_APP_SVC_SEARCH_URL}/rpc/search?document_type=${documentType}&return_facets=true&skip=${skip}&limit=${limit}`,
-    {
-      method: "POST",
-      headers: postHeaders,
-      body: JSON.stringify({ query: searchKeyword, filters: filterQuery }),
-    }
-  )
-    .then((response) => response.json())
-    .then(
-      (data) => {
-        callbackFunc(data);
-      },
-      (error) => {
-        alert("An error occured while fetching the data.");
-        const errorData: searchResponseModel = {
-          count: -1,
-          hits: [],
-          facets: [],
-        };
-        callbackFunc(errorData);
-      }
-    );
+  let url = `${SEARCH_URL}/rpc/search`;
+  url += `?document_type=${documentType}&return_facets=true&skip=${skip}&limit=${limit}`;
+  const headers = await getHeaders();
+  const body = JSON.stringify({ query: searchKeyword, filters: filterQuery });
+  try {
+    const response = await fetch(url, {method: "POST", headers, body});
+    const data = await response.json();
+    callbackFunc(data);
+  } catch(error) {
+    console.error(error);
```

Comment:
In general, we do not log errors on the console, is there a reason why we are doing this?

Reply:
Just so that a developer can see the actual error. There is an alert() but it is unspecific (and a abit obtrusive). I think the error handling should be improved when we go to production anyway. Do you think there is a drawback when logging errors on the console?

Reply:
Anyway, I remove the logging from this PR. We should discuss the best way of handling client side errors separately and more generally. I guess we should also separate dev mode and prod mode then.